ERCS/erc-7764.md

```diff
@@ -0,0 +1,136 @@
+---
+eip: 7764
+title: Buyer-Seller Negotiable Pricing
+description: Allows buyers and sellers to freely negotiate and determine prices.
+author: S7iter (@admi-n)
+discussions-to: https://ethereum-magicians.org/t/erc-7764-buyer-seller-negotiable-pricing/20973
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-09-05
+---
+
+## Abstract
+
+This proposal introduces a new smart contract mechanism that allows buyers and sellers to freely negotiate and determine transaction prices on the Ethereum network. A new trading mode is added, where goods can be negotiated rather than only priced. This mechanism allows the seller to set an initial price, and the buyer can propose a new price through negotiation with the seller. Ultimately, both parties can reach an agreement and complete the transaction.
+
+## Motivation
+
+Current smart contracts and decentralized applications (DApps) typically use a fixed-price model, where prices are set rather than negotiated. This model has many disadvantages, which may limit market flexibility and liquidity. A free price negotiation mechanism between buyers and sellers can better reflect market demand. This proposal introduces the most critical party in the transaction: the buyer's personal needs and pricing intentions. It can promote more transaction functionalities and models, expand markets, facilitate transactions, and increase user participation and satisfaction.
+
+## Specification
+
+### Structure
+
+The contract consists of the following main structures:
+
+- **Offer**: Represents the item or token listed by the seller and its initial price.
+- **Negotiation**: Represents the new price proposed by the buyer and whether it is accepted by the seller.
+
+### State Variables
+
+- `mapping(uint256 => Offer) public offers;`: Stores detailed information on all listed items or tokens.
+- `mapping(uint256 => Negotiation) public negotiations;`: Stores detailed information on all price negotiations.
+- `uint256 public nextOfferId;`: Used to generate new `offerId`.
```

Comment:
Specifying variables like this leaks too much of the implementation details. You should only specify the externally visible behaviour of your interface (so just functions and events.)